From 3fc953510ab5f0f48596ab2abc2d0ecc1fb136cd Mon Sep 17 00:00:00 2001
From: ejaaskel <esa.jaaskela@suomi24.fi>
Date: Tue, 9 Dec 2025 14:38:23 +0300
Subject: [PATCH] policy/modules/admin/dmesg: Fixes for dmesg

Loosen up some policies for dmesg_t to allow normal operation of the
dmesg during boot. It seems like sysvinit has an initialization script
that calls dmesg that requires these permissions, as they are not
required anymore post-boot. 

Upstream-Status: Inappropriate [Sulka specific]
---
 policy/modules/admin/dmesg.te | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/policy/modules/admin/dmesg.te b/policy/modules/admin/dmesg.te
index f1da315a9..d62656a93 100644
--- a/policy/modules/admin/dmesg.te
+++ b/policy/modules/admin/dmesg.te
@@ -1,5 +1,10 @@
 policy_module(dmesg)
 
+gen_require(`
+	type bin_t;
+	class file { map };
+')
+
 ########################################
 #
 # Declarations
@@ -19,6 +24,9 @@ dontaudit dmesg_t self:capability sys_tty_config;
 
 allow dmesg_t self:process signal_perms;
 
+allow dmesg_t bin_t:file { map read execute };
+allow dmesg_t proc_t:file { read open getattr };
+
 kernel_read_kernel_sysctls(dmesg_t)
 kernel_read_ring_buffer(dmesg_t)
 kernel_clear_ring_buffer(dmesg_t)
-- 
2.34.1

